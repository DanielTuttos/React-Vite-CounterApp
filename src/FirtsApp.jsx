import PropTypes from 'prop-types';

export const FirtsApp = ({ title, subtitle, name }) => {

    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            {/* <code>{JSON.stringify(newMessage)}</code> */}
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    )
}

// obliga a enviar valores requeridos
FirtsApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,

}

FirtsApp.defaultProps = {
    name: "Daniel Romero",
    // title: 'no hay titulo',
    subtitle: 'no hay subtitulo'
}