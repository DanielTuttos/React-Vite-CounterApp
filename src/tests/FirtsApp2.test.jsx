import { render, screen } from "@testing-library/react";
import { FirtsApp } from "../FirtsApp";

describe('Tests in <FirtsApp />', () => {

    const title = "Hola que tal como estan";
    const subtitle = "Soy un subtitulo";


    test('must have match with snapshot', () => {
        const { container } = render(<FirtsApp title={title} />)
        expect(container).toMatchSnapshot();
    });

    test('must to show the message "Hola que tal como estan"', () => {
        render(<FirtsApp title={title} />)
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('must to show the title in h1', () => {
        render(<FirtsApp title={title} />)
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('Must to show the subtitle to send by props', () => {
        render(
            <FirtsApp
                title={title}
                subtitle={subtitle}
            />
        )
        expect(screen.getAllByText(subtitle).length).toBe(2);// verifica que vengan 2
    });

})