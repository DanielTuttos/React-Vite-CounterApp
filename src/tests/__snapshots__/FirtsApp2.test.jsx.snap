// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirtsApp /> must have match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola que tal como estan
  </h1>
  <p>
    no hay subtitulo
  </p>
  <p>
    no hay subtitulo
  </p>
  <p>
    Daniel Romero
  </p>
</div>
`;
